### PR TITLE
[yarn] Update yarn spec with "global" subcommands

### PR DESCRIPTION
**Pull request by EliasGcf**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Update yarn spec

**Additional info:**

Add the `global` subcommands below
- `add` (with generator)
- `bin`
- `dir`
- `ls`
- `list`
- `remove` (with generator)
- `upgrade`
- `upgrade-interactive`

### CODE REVIEW

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for this PR! Few suggestions to resolve before merging it

File: dev/yarn.ts [hunk 1]
```diff
@@ -91,6 +91,39 @@ const packageList: Fig.Generator = {
...
+const getGlobalPackagesGenerator: Fig.Generator = {
+  script: 'cat "$(yarn global dir)/package.json"',
+  postProcess: (out, context) => {
+    if (out.trim() == "") return [];
+
+    try {
+      const packageContent = JSON.parse(out);
+      const dependencyScripts = packageContent["dependencies"] || {};
+      const devDependencyScripts = packageContent["devDependencies"] || {};
+      if (dependencyScripts || devDependencyScripts) {
```
[the remark this answers is not in the document]
Comment (EliasGcf) [reply dev/yarn.ts:104]: I copy this code from `packageList`, do you want I remove this `if` from there to?

**Comment by EliasGcf** [comment]
Done!

File: dev/yarn.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuiiBz) [reply dev/yarn.ts]: That would be great!

**Comment by QuiiBz** [comment]
> Done!

Thanks! Could you remove the other `if` as discussed here? https://github.com/withfig/autocomplete/pull/440#discussion_r683397116

File: dev/yarn.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (EliasGcf) [reply dev/yarn.ts]: As you wish!

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**